content/blog/2023-07-19-semgrep.md

```diff
@@ -0,0 +1,132 @@
+---
+title: Linting PromQL expressions using semgrep
+created_at: 2023-07-23
+kind: article
+author_name: Michael Hoffmann (https://github.com/MichaHoffmann)
+---
+
+### Motivation
+
+Writing good alerting rules is surprisingly hard in my experience. Consider the following alerting rule:
+
+```
+expr: -|
+    sum(rate(http_requests_total{route="/health", status~="5.."}[5m])) 
+    /
+    sum(rate(http_requests_total{route="/health"}[5m])) 
+    > 0.1
+```
+
+which seems innocent, until you notice that you forgot to match the `tenant_id` label to your team and discover
```

Comment:
I think we need some intro that in your setup you use tenant_id extensively. If this was a Thanos blog post, that might be fine (although some mention why you have tenant_id would be lovely too).